pydy/codegen/ode_function_generators.py

```diff
@@ -885,11 +898,11 @@ def __init__(self, *args, **kwargs):
 
     def define_inputs(self):
 
-        if self.specifieds is None:
-            self.inputs = self.coordinates + self.speeds + self.constants
-        else:
-            self.inputs = (self.coordinates + self.speeds + self.specifieds
-                           + self.constants)
+        specifieds = []
+        if self.specifieds is not None:
+            specifieds = self.specifieds
+        self.inputs = chain(self.coordinates, self.speeds, specifieds,
```

Comment:
Note that line 724 still defines self.inputs as a list of sequences. We should have consistency.

The define_inputs() method is currently a public method and in the base class it produces a certain type, i.e. list of sequences. But this line is in the subclass. I'd prefer the super method and the method in this class to return the same type.